### DISCUSSION TITLE
Exception to "no copy-paste" rule for superset languages?

### DISCUSSION THREAD

**Discussion by fnands** [rouge-ruby/rouge#1993] [Q&A] [answered]
Hi, 

I want to start creating a Rouge lexer for [Mojo](https://docs.modular.com/mojo/).  

Mojo aims to be a superset of Python, so valid Python code (should be, not all of it is yet) valid Mojo code. 

So does anyone object to me doing a copy-paste job in this case?

**Comment by jneen** [comment, marked as answer]
You can inherit the Python lexer and add additional rules.

**Reply by fnands** [reply]
Oh yeah great, that makes a lot more sense. 
Will do it that way then. Thx!
